src/pingdom-service-monitor.go

```diff
@@ -0,0 +1,148 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/russellcardullo/go-pingdom/pingdom"
+	"log"
+	"net/url"
+	"strconv"
+)
+
+// PingdomService interfaces with monitor-proxy
+type PingdomService struct {
+	apiKey        string
+	username      string
+	password      string
+	url           string
+	alertContacts string
+}
+
+func (monitor *PingdomService) GetAll() []Monitor {
+	var monitors []Monitor
+	client := pingdom.NewClient(monitor.username, monitor.password, monitor.apiKey)
+	checks, err := client.Checks.List()
+	if err != nil {
+		log.Println("error recieved listing checks: " + err.Error())
+	}
+	for _, mon := range checks {
+		newMon := Monitor{
+			url:  mon.Hostname,
+			id:   fmt.Sprintf("%v", mon.ID),
+			name: mon.Name,
+		}
+		monitors = append(monitors, newMon)
+	}
+	return monitors
+}
+
+func (monitor *PingdomService) Add(m Monitor) {
+	client := pingdom.NewClient(monitor.username, monitor.password, monitor.apiKey)
+	checkConfig := m.setupPingdomConfigsFromAnnotations(m.annotations)
+	_, err := client.Checks.Create(&checkConfig)
+	if err != nil {
+		log.Println("error adding service:", err.Error())
+	}
+	log.Println("Added monitor for: ", m.name)
+}
+
+func (monitor *PingdomService) Update(m Monitor) {
+	client := pingdom.NewClient(monitor.username, monitor.password, monitor.apiKey)
+	checkConfig := m.setupPingdomConfigsFromAnnotations(m.annotations)
+	intID, _ := strconv.Atoi(m.id)
+	resp, err := client.Checks.Update(intID, &checkConfig)
+	if err != nil {
+		log.Println("Error updating service:", err.Error())
+	}
+	log.Println("Update service response:", resp)
+}
+
+func (monitor *PingdomService) GetByName(name string) (*Monitor, error) {
+	var match *Monitor
+	allMonitors := monitor.GetAll()
+	for _, mon := range allMonitors {
+		if mon.name == name {
+			match = &mon
+		}
+	}
+	if match == nil {
+		return match, fmt.Errorf("Unable to locate service with name %v", name)
+	}
+	return match, nil
+}
+
+func (monitor *PingdomService) Remove(m Monitor) {
+	client := pingdom.NewClient(monitor.username, monitor.password, monitor.apiKey)
+	intID, _ := strconv.Atoi(m.id)
+	log.Println("Deleting monitoring: ", m.name)
+	client.Checks.Delete(intID)
+}
+
+func (monitor *PingdomService) Setup(apiKey string, url string, alertContacts string, username string, password string) {
+	monitor.apiKey = apiKey
+	monitor.url = url
+	monitor.alertContacts = alertContacts
+	monitor.username = username
+	monitor.password = password
+}
+
+func (monitor *Monitor) setupPingdomConfigsFromAnnotations(annotations map[string]string) (checkConfig pingdom.HttpCheck) {
```

Comment:
You have made annotations a part of `Monitor` and yet you're passing in those annotations.
Also I don't think this method should be a part of `Monitor` struct. Maybe make it a package private method of this service and then pass in the annotations from the monitor object. That way `Monitor` won't contain any specific implementation